build.zig

```diff
@@ -79,6 +79,75 @@ fn linkLibraries(b: *std.Build, exe: *std.Build.Step.Compile, useLocalDeps: bool
 	}
 }
 
+pub fn makeModFeature(step: *std.Build.Step, _: std.Build.Step.MakeOptions) anyerror!void {
+	var out: std.ArrayListUnmanaged(u8) = .{};
+
+	var assetsDir = try std.fs.cwd().openDir("mods", .{.iterate = true});
+	defer assetsDir.close();
+
+	var iterator = assetsDir.iterate();
+	while(try iterator.next()) |addonEntry| {
+		if(addonEntry.kind != .directory) continue;
+
+		var addon = try assetsDir.openDir(addonEntry.name, .{});
+		defer addon.close();
+
+		var rotationDir = addon.openDir(step.name, .{.iterate = true}) catch continue;
+		defer rotationDir.close();
+
+		var rotationIterator = rotationDir.iterate();
+		while(try rotationIterator.next()) |rotationEntry| {
+			if(rotationEntry.kind != .file) continue;
+			if(!std.mem.endsWith(u8, rotationEntry.name, ".zig")) continue;
+
+			try out.appendSlice(step.owner.allocator, step.owner.fmt(
+				"pub const @\"{s}:{s}\" = @import(\"{s}/{s}/{s}\");\n",
+				.{
+					addonEntry.name,
+					rotationEntry.name[0 .. rotationEntry.name.len - 4],
+					addonEntry.name,
+					step.name,
+					rotationEntry.name,
+				},
+			));
+		}
+	}
+
+	const file_path = try std.fs.path.join(step.owner.allocator, &.{"mods", step.owner.fmt("{s}.zig", .{step.name})});
+	var file = try std.fs.cwd().createFile(file_path, .{.truncate = true});
+	defer file.close();
+
+	try file.writeAll(out.items);
+	out.deinit(step.owner.allocator);
```

Comment:
Should be in a defer next to the creation.
Down here it's leaking in 10 places.